### PR TITLE
gix-transport: add SHA-256 object-format negotiation test coverage

### PR DESCRIPTION
**Pull request by 10ne1**
**What**

This PR is a companion to #2642 (which does the protocol-level SHA-256 negotiation in `gix-protocol` and `gix`) and covers the layer below it: proving `gix-transport` relays the server  `object-format=sha256` capability through the V2 handshake.

This just adds three tests and doesn't require any "production" changes, because `gix-transport` is hash-agnostic by design and treats capabilities as simple key=value pairs, so the sha256 advertisments flow through unmodified.

Part of #281 (Batch 3, "Protocol and Transport"), transport half.

**Testing**

  - reqwest: `cargo nextest run -p gix-transport --no-default-features --features blocking-client,http-client-reqwest,http-client-insecure-credentials,maybe-async/is_sync --test blocking-transport`
  - curl: `cargo nextest run -p gix-transport --features http-client-curl,http-client-insecure-credentials,maybe-async/is_sync --test blocking-transport-http-only`
  - async: `cargo nextest run -p gix-transport --features async-client`

These are canned byte fixtures, so they run under the existing default-feature test runs, no `GIX_TEST_FIXTURE_HASH` or generated archives involved.

**AI disclosure**

This PR was written with the help of Claude Code, but was **not** vibe-coded. I'm not a fan of vibe-coding. I used AI to better understand the codebase myself (asked it a lot of questions), review both my code and the code generated by AI, used it especially to detect bugs and corner-cases and suggest fixes, however I very carefully reviewed & edited each code/comment lines, the commit messages and so on, until I was satisfied with the result.

### CODE REVIEW

File: gix-transport/tests/fixtures/v2/http-handshake-sha256.response [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+X-Frame-Options: DENY
+X-GitHub-Request-Id: 737D:544E:C932CB:113F404:5F3F3EE5
+
+000eversion 2
+0023agent=git/github-gdf51a71f0236
+000cls-refs
+0019fetch=shallow filter
+0012server-option
+0019object-format=sha256
+0000
```
[the remark this answers is not in the document]
Comment (10ne1) [reply gix-transport/tests/fixtures/v2/http-handshake-sha256.response:18]: I added a new commit to address this. Thanks!

File: gix-transport/tests/client/git.rs [hunk 2]
```diff
@@ -452,3 +452,42 @@ async fn handshake_v2_and_request_inner() -> crate::Result {
...
+            ("ls-refs", None),
+            ("fetch", Some("shallow")),
+            ("server-option", None),
+            ("object-format", Some("sha256")),
+        ]
+        .iter()
+        .map(|(k, v)| (k.as_bytes().into(), v.map(|v| v.as_bytes().into())))
+        .collect::<Vec<_>>(),
+        "the sha256 object-format advertised by the server is surfaced from the V2 handshake"
+    );
```
[the remark this answers is not in the document]
Comment (10ne1) [reply gix-transport/tests/client/git.rs:501]: Updated the initial commit to use values().

**Comment by 10ne1** [comment]
I reworded the commit message subjects to conform to the gitoxide rules, no other code changes.

**Comment by Byron** [comment]
Thanks a lot for driving this forward!

While for a moment I was thinking that this is probably "over"-testing it, I also thought that having those tests probably won't hurt.

Please note that this PR once again had to be superseded by https://github.com/GitoxideLabs/gitoxide/pull/2656 to capture my own commit. If it's no hassle for you, allowing me to push back would be appreciated.

Closing as superseded.
